File: frontend/src/types/announcement.ts
export type AnnouncementCategory = 'GENERAL' | 'POLICY' | 'ACTIVITY' | 'WELFARE' | 'URGENT';
export type AnnouncementPriority = 'LOW' | 'NORMAL' | 'HIGH' | 'URGENT';
export type AnnouncementStatus = 'DRAFT' | 'PUBLISHED' | 'ARCHIVED';
export type TargetType = 'ALL' | 'DEPARTMENT' | 'DIVISION' | 'EMPLOYEE_LEVEL';

export interface PagedResult<T> {
  items: T[];
  totalCount: number;
  page: number;
  pageSize: number;
  totalPages: number;
}

export interface AnnouncementTarget {
  id: number;
  targetType: TargetType;
  targetEntityId?: number | null;
  targetEntityName?: string | null;
}

export interface AnnouncementTargetInput {
  targetType: TargetType;
  targetEntityId?: number | null;
}

export interface Announcement {
  id: number;
  title: string;
  content: string;
  publishedAt?: string | null;
  expireAt?: string | null;
  createdByEmployeeId?: number | null;
  createdByEmployeeName?: string | null;
  status: AnnouncementStatus;
  isPinned: boolean;
  bannerImageUrl?: string | null;
  category: AnnouncementCategory;
  priority: AnnouncementPriority;
  createdAt: string;
  updatedAt?: string | null;
  targets: AnnouncementTarget[];
  readCount: number;
  totalTargetCount: number;
  isReadByCurrentUser?: boolean;
  readAtByCurrentUser?: string | null;
}

export interface CreateAnnouncementPayload {
  title: string;
  content: string;
  publishedAt?: string | null;
  expireAt?: string | null;
  isPinned?: boolean;
  bannerImageUrl?: string | null;
  category?: AnnouncementCategory;
  priority?: AnnouncementPriority;
  status?: AnnouncementStatus;
  targets?: AnnouncementTargetInput[];
}

export interface UpdateAnnouncementPayload {
  title: string;
  content: string;
  publishedAt?: string | null;
  expireAt?: string | null;
  isPinned?: boolean;
  bannerImageUrl?: string | null;
  category?: AnnouncementCategory;
  priority?: AnnouncementPriority;
  status?: AnnouncementStatus;
  targets?: AnnouncementTargetInput[];
}

export interface ReadReceipt {
  employeeId: number;
  employeeCode: string;
  employeeName: string;
  departmentName?: string | null;
  readAt: string;
}

export interface AnnouncementReadStats {
  announcementId: number;
  title: string;
  totalTargetEmployees: number;
  readCount: number;
  readPercentage: number;
  receipts: ReadReceipt[];
}

export interface AnnouncementFilterParams {
  search?: string;
  status?: string;
  category?: string;
  priority?: string;
  isPinned?: boolean;
  page?: number;
  pageSize?: number;
}

File: backend/src/Hrms.Domain/Entities/Announcement.cs
using Hrms.Domain.Common;

namespace Hrms.Domain.Entities;

/// <summary>
/// ข่าวสารและประกาศประชาสัมพันธ์องค์กร
/// แมปกับตาราง hrms.announcement
/// </summary>
public class Announcement : BaseEntity
{
    public string Title { get; set; } = string.Empty;
    public string Content { get; set; } = string.Empty;
    public DateTime? PublishedAt { get; set; }
    public DateTime? ExpireAt { get; set; }
    public long? CreatedByEmployeeId { get; set; }
    public string Status { get; set; } = "DRAFT"; // DRAFT, PUBLISHED, ARCHIVED
    public bool IsPinned { get; set; } = false;
    public string? BannerImageUrl { get; set; }
    public string Category { get; set; } = "GENERAL"; // GENERAL, POLICY, ACTIVITY, WELFARE, URGENT
    public string Priority { get; set; } = "NORMAL"; // LOW, NORMAL, HIGH, URGENT
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }

    // Navigation Properties
    public virtual Employee? CreatedByEmployee { get; set; }
    public virtual ICollection<AnnouncementTarget> Targets { get; set; } = new List<AnnouncementTarget>();
    public virtual ICollection<AnnouncementRead> Reads { get; set; } = new List<AnnouncementRead>();
}
